vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java

```diff
@@ -0,0 +1,438 @@
+     *
+     * @param autoselect
+     *            <code>true</code> to set auto select on, <code>false</code>
+     *            otherwise
+     */
+    @Override
+    public void setAutoselect(boolean autoselect) {
+        super.setAutoselect(autoselect);
+    }
+
+    /**
+     * Gets the visibility state of the button which clears the number field.
+     *
+     * @return <code>true</code> if the button is visible, <code>false</code>
+     *         otherwise
+     */
+    public boolean isClearButtonVisible() {
+        return isClearButtonVisibleBoolean();
+    }
+
+    /**
+     * Set to <code>false</code> to hide the clear button which clears the
+     * number field.
+     *
+     * @param clearButtonVisible
+     *            <code>true</code> to set the button visible,
+     *            <code>false</code> otherwise
+     */
+    @Override
+    public void setClearButtonVisible(boolean clearButtonVisible) {
+        super.setClearButtonVisible(clearButtonVisible);
+    }
+
+    /**
+     * Returns the value that represents an empty value.
+     */
+    @Override
+    public T getEmptyValue() {
+        return null;
+    }
+
+    /**
+     * Sets the value of this number field. If the new value is not equal to
+     * {@code getValue()}, fires a value change event.
+     *
+     * @param value
+     *            the new value
+     */
+    @Override
+    public void setValue(T value) {
+        super.setValue(value);
+    }
+
+    /**
+     * Returns the current value of the number field. By default, the empty
+     * number field will return {@code null} .
+     *
+     * @return the current value.
+     */
+    @Override
+    public T getValue() {
+        return super.getValue();
+    }
+
+    @Override
+    protected void setMin(double min) {
+        super.setMin(min);
+        this.min = min;
+    }
+
+    @Override
+    protected double getMinDouble() {
+        return min;
+    }
+
+    @Override
+    protected void setMax(double max) {
+        super.setMax(max);
+        this.max = max;
+    }
+
+    @Override
+    protected double getMaxDouble() {
+        return max;
+    }
+
+    @Override
+    protected void setStep(double step) {
+        super.setStep(step);
+        this.step = step;
+    }
+
+    @Override
+    protected double getStepDouble() {
+        return step;
+    }
+
+    @Override
+    protected void setModelValue(T newModelValue, boolean fromClient) {
+        super.setModelValue(newModelValue, fromClient);
+        validate();
+    }
+
+    @Override
+    public void setInvalid(boolean invalid) {
+        super.setInvalid(invalid);
+    }
+
+    @Override
+    public boolean isInvalid() {
+        return isInvalidBoolean();
+    }
+
+    /**
+     * Performs server-side validation of the current value. This is needed
+     * because it is possible to circumvent the client-side validation
+     * constraints using browser development tools.
+     */
+    @Override
+    protected void validate() {
```

Comment:
Moved validation to this method to avoid repeating `setInvalid(isInvalid(getValue()))`